lib/config/index.js

```diff
@@ -74,6 +74,71 @@ exports.getLoaderConfig = function() {
   return cfg;
 };
 
+function updateNodelibTarget(target) {
+  target.registry = 'npm';
+  target.name = target.name.replace('github:jspm/nodelibs-', 'npm:jspm-nodelibs-');
+  target.package = target.package.replace('jspm/nodelibs-', 'jspm-nodelibs-');
+  target.exactPackage = target.exactPackage.replace('jspm/nodelibs-', 'jspm-nodelibs-');
+  target.exactName = target.exactName.replace('github:jspm/nodelibs-', 'npm:jspm-nodelibs-');
+  target.exactNameEncoded = target.exactNameEncoded.replace('github:jspm/nodelibs-', 'npm:jspm-nodelibs-');
+  target.setVersion('0.2.0');
+  return target;
+}
+
+function processNodelibs(loader, pjson) {
+  var installRequired = false;
+  Object.keys(loader.baseMap).forEach(function (key) {
+    var target = loader.baseMap[key];
+
+    if (target.name.substr(0, 21) == 'github:jspm/nodelibs-') {
+      target = updateNodelibTarget(target);
+      installRequired = true;
+    }
+  });
+
+  Object.keys(loader.depMap).forEach(function (key) {
+    var curMap = loader.depMap[key];
+
+    Object.keys(curMap).forEach(function (key) {
+      var target = curMap[key];
+
+      if (target.name.substr(0, 21) == 'github:jspm/nodelibs-') {
+        if (key == target.name.substr(21))
+          delete curMap[key];
+        else
+          installRequired = true;
```

Comment:
The original version of this code was designed to have two separate branches:

* If the submap name matches the node core lib name (`fs: github:jspm-nodelibs-fs@...`) then it deletes the key entirely. This is for the beta upgrade path so we turn Node core dependencies from normal dependencies into peer dependencies.
* If the submap name doesn't match (the else above in the original code) then update the target version (`custom-fs-name: github:nodelibs-fs@...`). That's a very rare case for jspm-only workflows but what this was designed for in the original code to handle this upgrade comprehensively.

In the adjusted code above `installRequired` is now set on the else path only, and `target` is always updated on all paths due to the lack of curly braces on the conditional.

So there are two problems here:

1. The code above is malformed, at least from a tabbing perspective.
2. We probably do need another flag to indicate the difference between the beta upgrade path and the nodelibs upgrade path. In the first case we want the original behaviour to be maintained. In the second case we just need to do replacement (the target update line below), and probably don't even need the conditional above.

Let me know if that makes sense.